internal/pkg/v2/infrastructure/redis/notification.go

```diff
@@ -213,3 +213,142 @@ func notificationsByCategoriesAndLabels(conn redis.Conn, offset int, limit int,
 	}
 	return convertObjectsToNotifications(objects)
 }
+
+// notificationAndTransmissionStoreKeys return the store keys of the notification and transmission that are older than age.
+func notificationAndTransmissionStoreKeys(conn redis.Conn, age int64) ([]string, []string, errors.EdgeX) {
+	expireTimestamp := common.MakeTimestamp() - age
+
+	ncStoreKeys, err := redis.Strings(conn.Do(ZRANGEBYSCORE, NotificationCollection, 0, expireTimestamp))
+	if err != nil {
+		return nil, nil, errors.NewCommonEdgeX(errors.KindDatabaseError, fmt.Sprintf("retrieve nitification storeKeys by %s failed", NotificationCollection), err)
+	}
+
+	var transStoreKeys []string
+	for _, ncStoreKey := range ncStoreKeys {
+		keys, err := redis.Strings(conn.Do(ZRANGE, CreateKey(TransmissionCollectionNotificationId, idFromStoredKey(ncStoreKey)), 0, -1))
+		if err != nil {
+			return nil, nil, errors.NewCommonEdgeX(errors.KindDatabaseError, "fail to retrieve transmission storeKeys", err)
+		}
+		transStoreKeys = append(transStoreKeys, keys...)
+	}
+
+	return ncStoreKeys, transStoreKeys, nil
+}
+
+// asyncDeleteReadingsByIds deletes all notifications with given storeKeys. This function is implemented to be run as a
```

Comment:
change `asyncDeleteReadingsByIds` to `asyncDeleteNotificationByStoreKeys`

Reply:
Thank you for catch that. Modified.